### ISSUE TITLE
Object Detection

### ISSUE THREAD

**Issue by ghost** [razvancaramalau/Visual-Transformer-for-Task-aware-Active-Learning#3]
Hello, Thank you for sharing works!

I try to reproduce the experiment on VOC with [ssd.pytorch](https://github.com/amdegroot/ssd.pytorch).
In the paper,
`The visual transformer bottleneck from our joint-learning selection is positioned only on the confidence head of the SSD network.`

Can you explain this in detail? There are confidence heads from 6 feature maps.

**Comment by razvancaramalau** [comment]
Hi there,
The input to the transformer bottleneck comes only from the confidences for all categories (C) as 8732 x (C B) where B is the batch size. So, the 8732 are the concatenated feature maps from all 6 heads. Thanks.

**Comment by ghost** [comment]
How to pass output of transformer to discriminator (sampler)?
After transformer, the shape of input tensor to the discriminator is `[B, (8732 x C)]`?

**Comment by razvancaramalau** [comment]
That's correct [B, (8732 x C)].

**Comment by ghost** [comment]
I try to add the visual transformer bottleneck to [here](https://github.com/amdegroot/ssd.pytorch/blob/master/ssd.py#L97)
```
conf = torch.cat([o.view(o.size(0), -1) for o in conf], 1) # shape: [B, (8732 x C)]
conf = rearrange(conf, 'b (conf c) -> conf (b c)')
conf = transformer(conf)
conf = rearrange(conf, 'conf (b c) -> b conf c', b=B, c=C).contiguous()

if self.phase == "test":
       output = self.detect(
                loc.view(loc.size(0), -1, 4),                   # loc preds
                self.softmax(conf),                # conf preds
                self.priors.type(type(x.data))                  # default boxes
            )
        else:
            output = (
                loc.view(loc.size(0), -1, 4),
                conf,
                self.priors
            )
        return output

```
But the `conf` became `nan` after few steps. Did I do anything wrong?
I used [discriminator](https://github.com/razvancaramalau/Visual-Transformer-for-Task-aware-Active-Learning/blob/master/models/query_models.py#L219)(z_dim=183372) and [transformer](https://github.com/razvancaramalau/Visual-Transformer-for-Task-aware-Active-Learning/blob/master/models/resnet.py#L302)(dim=672, depth=1, heads=1, mlp_dim=128, dropout=0.1)

**Comment by razvancaramalau** [comment]
Yes, that looks correct. I've experienced the same in the beginning. Adjusting the learning rate and having a few runs stabilizes the gradients after the first 10 iterations. I was actually thinking of even shifting to another optimizer.

**Comment by ghost** [comment]
So far, I've used hyperparamters in the learning loss paper.

According to your advice, I lowered learning rate from 1e-3 to 1e-4 and then `conf` didn't become `nan`.
But mAP is 6.77 at the first cycle (# annotated samples: 1,000)
How to adjust the learning rate?
